ce/api/percentileanomaly.py

```diff
@@ -183,89 +279,69 @@ def canonical_timestamp(timestamp, timescale, timeidx):
 
                     if row['climatology'] == climatology:
-                        add_to_nested_dict(projected_data,
-                                           row["mean"],
-                                           row["timescale"],
-                                           ctimestamp,
-                                           row["model"])
+
+                        obj = create_data_object(row["mean"], row["timescale"],
+                                                ctimestamp, row["timeidx"], row["model"])
+                        projected_data.append(obj)
+
                     elif (calculate_anomaly and
                             row['model'] == baseline_model and
                             row['climatology'] == baseline_climatology):
-                        add_to_nested_dict(baseline_data, row['mean'],
-                                           row['timescale'], ctimestamp)
+                        obj = create_data_object(row["mean"], row["timescale"], ctimestamp, row["timeidx"])
+                        baseline_data.append(obj)
+
+        projected_data.sort(key=lambda e: e['dateidx'])
+        merge_duplicates(projected_data)
+        baseline_data.sort(key=lambda e: e['dateidx'])  
+        merge_duplicates(baseline_data)
+
+
 
         # calculate percentiles and anomalies
-        for timescale in projected_data:
-            for timestamp in projected_data[timescale]:
-                # determine the baseline, if applicable
-                if calculate_anomaly:
-                    baseline = None
-                    date_format = '%Y-%m-%d %H:%M:%S'
-                    projected_date = datetime.strptime(timestamp, date_format)
-                    if timescale not in baseline_data:
-                        abort(500, "Missing baseline data: {} {} {}".format(
-                            baseline_model, baseline_climatology, timescale))
-                    for t in baseline_data[timescale]:
-                        baseline_date = datetime.strptime(t, date_format)
-                        if baseline_date.month == projected_date.month:
-                            if not baseline:
-                                baseline = float(baseline_data[timescale][t])
-                            else:
-                                abort(500,
-                                      ("Multiple matching baseline datasets ",
-                                       "for {} {} {} {}").format(
-                                            baseline_model,
-                                            baseline_climatology,
-                                            timescale, timestamp))
-                    if baseline is None:
-                        abort(500,
-                              ("No baseline match available for ",
-                               "{} {} {} {}").format(
-                                   baseline_model, baseline_climatology,
-                                   timescale, timestamp))
```

Comment:
I had to omit `abort(500, "Missing baseline data: {} {} {}".format(baseline_model, baseline_climatology, timescale))` error message since there is no way to check if the `timescale` is in `baseline_data` before the `for` loop. I don't see any problem not having this message as this error is a subset of the latter error:
```
                        abort(500,
                              ("No baseline match available for ",
                               "{} {} {} {}").format(
                                   baseline_model, baseline_climatology,
                                   timescale, timestamp))
```